### DISCUSSION TITLE
Pekko-gRPC support on Apple Silicon (M1 / M2)

### DISCUSSION THREAD

**Discussion by jonathan-neufeld-asurion** [apache/pekko#375] [Q&A] [answered]
Is there a plan to support **Apple Silicon** for Pekko-gRPC?  Protocol buffers does not generate the class files in versions prior to Akka @ v2.7 + `sbt-akka-grpc` plugin @ v2.2.1, there is a missing native dependency:

```
[error] stack trace is suppressed; run last model / protocExecutable for the full output
[error] (model / protocExecutable) lmcoursier.internal.shaded.coursier.error.FetchError$DownloadingArtifacts: Error fetching artifacts:
[error] https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.13.0/protoc-3.13.0-osx-aarch_64.exe: not found: https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.13.0/protoc-3.13.0-osx-aarch_64.exe
[error] Total time: 1 s, completed Jun 7, 2023, 8:00:31 PM
```

Related issue: https://github.com/grpc/grpc-java/issues/7690

Since Pekko is branched from Akka 2.6 I am concerned that we will run into the above issue with Pekko and protocol buffers.  We resolved it by upgrading Akka to v2.7 and the Akka-gRPC SBT plugin to v2.2.1.

**Comment by pjfanning** [comment]
protoc 3.13.0 doesn't support osx-aarch but newer versions like 3.23.2 seem to.

Have a look in https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.23.2/

Could you try modifying your checkout to use this version instead to see how the build progresses?

**Reply by jonathan-neufeld-asurion** [reply, marked as answer]
Using the plugin at version 2.1.6 was the trick, it does not require v2.7 of Akka.

**Comment by mdedetrich** [comment]
I just wanted to mention that I also use Apple M1 as my primary development machine and don't have issues with Pekko grpc
